### ISSUE TITLE
Only last snapshot is backeded up in 'manual' mode

### ISSUE THREAD

**Issue by TitusLabienus** [zrepl/zrepl#187]
When the backup mechanism encounters a filesystem that has not been created yet on the backup side, only the last snapshot is transferred.

### Steps to reproduce

**Preparation**

[YAML config file](https://github.com/zrepl/zrepl/files/3293623/zrepl.yml.txt)

```
# zfs list -o name -r -t filesystem,snapshot
NAME
DestinationPool
SourcePool
SourcePool/MyFilesystem
SourcePool/MyFilesystem@1
SourcePool/MyFilesystem@2
SourcePool/MyFilesystem@3
SourcePool/MyFilesystem@4
# systemctl start zrepl
# zrepl signal wakeup Backup
# zfs list -o name -r -t filesystem,snapshot
```
**Expected Result**
```
NAME
DestinationPool
DestinationPool/MyBackup
DestinationPool/MyBackup/SourcePool
DestinationPool/MyBackup/SourcePool/MyFilesystem
DestinationPool/MyBackup/SourcePool/MyFilesystem@1
DestinationPool/MyBackup/SourcePool/MyFilesystem@2
DestinationPool/MyBackup/SourcePool/MyFilesystem@3
DestinationPool/MyBackup/SourcePool/MyFilesystem@4
SourcePool
SourcePool/MyFilesystem
SourcePool/MyFilesystem@1
SourcePool/MyFilesystem@2
SourcePool/MyFilesystem@3
SourcePool/MyFilesystem@4
```
**Result**
```
NAME
DestinationPool
DestinationPool/MyBackup
DestinationPool/MyBackup/SourcePool
DestinationPool/MyBackup/SourcePool/MyFilesystem
DestinationPool/MyBackup/SourcePool/MyFilesystem@4
SourcePool
SourcePool/MyFilesystem
SourcePool/MyFilesystem@1
SourcePool/MyFilesystem@2
SourcePool/MyFilesystem@3
SourcePool/MyFilesystem@4
```

**Comment by problame** [comment]
Thanks for the report.
This is not a bug but the current hard-coded policy for new filesystems:

https://github.com/zrepl/zrepl/blob/000d8bba6656e19ed00f57b9c3d2b1b9baf3f757/replication/logic/replication_logic.go#L232

If you would like to have this policy configurable, please propose a config syntax extension to handle different policies and propose policies as you wish, we'll discuss them here. Keep different user use-cases in mind.

**Comment by problame** [comment]
Update: #217 and #218 are being worked on and will bring infrastructure for configurable replication behavior in the config:

https://github.com/zrepl/zrepl/blob/f3c29b7b7289422d8593183cf62921b4504b005a/replication/logic/replication_logic.go#L53-L56

The initial replication policy could become just another field in that struct, and be wired to the config parser similar to `PlannerPolicy.EncryptedSend`

**Comment by 3nprob** [comment]
I guess this can be closed now that both #217 and #218 are landed?

**Comment by problame** [comment]
No, this feature request is not yet addressed: it asks for a config option to configure what should happen if there is no file system on the receiving side: send the most recent snapshot vs. send all snapshots.

**Comment by Greek64** [comment]
I'm actually surprised this "feature" is not more requested!
My guess is that most of the people that start using zrepl also just start snapshoting/replicating, and thus they usually do not have previous snapshots.
But why wouldn't you want to sent **all** your snapshots to the Backup server? Isn't that the whole purpose of a Backup? 
In other words I'm just surprised this isn't the default.

**Comment by shibattor** [comment]
I agree, I assumed it would behave in this way.
